rasa/core/schemas/domain.yml

```diff
@@ -27,6 +27,40 @@ mapping:
   responses:
     type: "map"
     allowempty: True
+    mapping:
+      regex;(.+):
+        type: "seq"
+        required: False
+        nullable: False
+        func: require_response_keys
+        sequence:
+        - type: "map"
+          required: True
+          allowempty: False
+          mapping:
+            text:
+              type: "str"
+            image:
+              type: "str"
+            custom:
+              type: "map"
+              allowempty: True
+            buttons:
+              type: "seq"
+              sequence:
+              - type: "map"
+                mapping:
+                  title:
+                    type: "str"
+                  payload:
+                    type: "str"
+            channel:
+              type: "str"
+            # allow custom keys
```

Comment:
is it necessary if we already have a very flexible `custom` key?
Asking because users can make a type, like `imag` instead of `image` and that would pass the verification.

Reply:
yeah I'd rather remove it too, but some tests failed because of this, for instance [this one](https://github.com/RasaHQ/rasa/blob/master/tests/nlu/training_data/formats/test_rasa_yaml.py#L311). Where can I find the definition of the format and what's officially supported?

Reply:
clarification from tom, who implemented that:

> keys are exactly the same as for domain responses (utter templates)

So that's `text` `buttons` `image` `custom` `channel`.

Reply:
I believe you can adjust the failing test and remove invalid examples.

Reply:
yayyyyy